### DISCUSSION TITLE
How to add/insert cst nodes to modules deep body

### DISCUSSION THREAD

**Discussion by un4gt** [Instagram/LibCST#1132] [Q&A]
Mentioned in this document [provide-a-config-when-generating-code-from-templatess](https://libcst.readthedocs.io/en/latest/best_practices.html#provide-a-config-when-generating-code-from-templates) how to add a statement to the end of a module，but what if I want to add to the middle?

![image](https://github.com/Instagram/LibCST/assets/81068011/76b620f1-465b-4c7d-9c69-9ac485c4c5f2)

I try to made this by python's builtin ast modules, but blank lines and comments will be deleted.

Looking forward to your answers, thank you

**Comment by disouzam** [comment]
I am also looking for an answer! I know a little about codemods, transformers but I am still a little confused!
